pkg/assembler/backends/neo4j/hasSLSA.go

```diff
@@ -36,22 +36,22 @@ const (
 )
 
 func (c *neo4jClient) HasSlsa(ctx context.Context, hasSLSASpec *model.HasSLSASpec) ([]*model.HasSlsa, error) {
+	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
+	defer session.Close()
+
 	err := checkHasSLSAInputs(hasSLSASpec)
 	if err != nil {
 		return nil, err
 	}
 
 	queryAll := false
-	if hasSLSASpec.Package == nil && hasSLSASpec.Source == nil && hasSLSASpec.Artifact == nil {
+	if hasSLSASpec.Subject.Package == nil && hasSLSASpec.Subject.Source == nil && hasSLSASpec.Subject.Artifact == nil {
```

Comment:
this can be changed to just `hasSLSASpec.Subject == nil` to query all

Reply:
I think I need to handle that case separately. I still have to do e2e testing after all ingestion is done. I'm planning to test/change that there.